### ISSUE TITLE
Style/ArrayCoercion does not play nicely with Hashes

### ISSUE THREAD

**Issue by adamcooke** [rubocop-hq/rubocop#8783]
## Expected behaviour

The Style/ArrayCoercion is not safe because it converts hashes to arrays.

## Actual behaviour

```
values = [values] unless values.is_a?(Array)
# is corrected to
values = Array(values)
```

Which leads to values being:

```
values = {a: 'b'}
values = Array(values) # => [[:a, "b"]] but it should be [{a: 'b'}]
```

## RuboCop version

```
$ bundle exec rubocop -V
0.91.1 (using Parser 2.7.1.4, rubocop-ast 0.4.2, running on ruby 2.6.5 x86_64-linux)
```

**Comment by marcandre** [comment]
Indeed. See #8391 and #8334

We're waiting for @bbatsov to decide to retire this Cop, or at least have a config so it targets `[*val]` but not `val = [val] unless val.is_a?(Array)` 😅

**Comment by bbatsov** [comment]
The main reason I didn't rush into a decision is that the cop has been around for a while and there are relatively few complaints which leads me to believe it's useful in most cases. If only Ruby's internals were more consistent. 😆 

I'm leaning towards simply disabling the cop, so that people can still run it on demand when needed, but some thoughts from others in @rubocop-hq/rubocop-core are welcome.

**Comment by koic** [comment]
I agree to disable the cop by default. Because users who want to detect the style guide rule in exchange for false positives may want to enable it.
